boards/beagle/pocketbeagle_2/pocketbeagle_2_am6232_m4-pinctrl.dtsi

```diff
@@ -14,4 +14,12 @@
 	mcu_uart0_tx_default: mcu_uart0_tx_default {
 		pinmux = <K3_PINMUX(0x0018, PIN_OUTPUT, MUX_MODE_0)>;
 	};
+
+	P2_09_D4_gpio: P2-09-D4-gpio-pins {
```

Comment:
To me, D4 is a comment and not something a user should need to look-up/remember. It seems like all the header pins should be defined and set all needed pinmuxes for that pin.

Reply:
This is the style we are following in [beagle devicetrees](https://github.com/beagleboard/BeagleBoard-DeviceTrees/blob/v6.12.x-Beagle/src/arm64/ti/k3-am6232-pocketbeagle2-pinmux.dtsi).
The reason is a lot of pins can be muxed in multiple ways, even for the same functionality. And it is possible to damage the hardware if you do it incorrectly.
Even for just MCU domain, `P2.11` can be muxed as `MCU_SPI0_CS2` and `MCU_SPI1_CS2`. Hence, we need to keep it part of the name.